data/censusProxy.go

```diff
@@ -0,0 +1,69 @@
+package data
+
+import (
+	"bytes"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type CensusTransport struct {
+	CacheRepository *CacheRepository
+}
+
+func (t *CensusTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	response, err := http.DefaultTransport.RoundTrip(r)
+	if err != nil {
+		// Check cache
+		cachedVal, _ := t.CacheRepository.GetCache(GetCensusReqURI(r))
+		if cachedVal != nil {
+			rBody := bytes.NewBuffer(cachedVal)
+			resp := SetCensusResponse(rBody)
+			resp.Header.Set("Content-Type", "application/json")
+			return resp, nil
+		}
+		// If cache is empty, return error message
+		rBody := bytes.NewBufferString("Error retrieving data")
+		resp := SetCensusResponse(rBody)
+		resp.Header.Set("Content-Type", "plain/text")
+		return resp, nil
```

Comment:
I would wrap this code, lines 18-29, in a function with the signature:
```
RetrieveCached(r *http.Request) (*http.Response, error)
```
Then the body of this conditional becomes
```
return RetrieveCache(r)
```